frontend/src/components/Modal/AssignDriverModal.tsx

```diff
@@ -0,0 +1,69 @@
+import React, { useRef, useEffect, useState } from 'react';
+import { Passenger, Driver } from '../../types/index';
+import styles from './assigndrivermodal.module.css';
+
+type AssignModalProps = {
+  index: number;
+  isOpen: boolean;
+  close: () => void;
+  ride: Passenger;
+}
+
+type DriverRowProps = {
+  firstName: string;
+  imageURL: string;
+}
+
+const DriverRow = ({ firstName, imageURL }: DriverRowProps) => (
+  <div className={styles.driverRow}>
+    <img className={styles.driverImage} src={imageURL} alt="Avatar"></img>
+    <p className={styles.driverName}>{firstName}</p>
+  </div>
+);
+
+const AssignDriverModal = ({ index, isOpen, close, ride }: AssignModalProps) => {
+  // source: https://stackoverflow.com/questions/32553158/detect-click-outside-react-component
+  function useOutsideAlerter(ref: any) {
+    useEffect(() => {
+      function handleClickOutside(event: any) {
+        if (ref.current && !ref.current.contains(event.target)) {
+          close();
+        }
+      }
+
+      document.addEventListener('mousedown', handleClickOutside);
+      return () => {
+        document.removeEventListener('mousedown', handleClickOutside);
+      };
+    }, [ref]);
+  }
+
+  const [drivers, setDrivers] = useState<Driver[]>([]);
+
+  const getExistingDrivers = async () => {
+    const driverData = await fetch('/drivers')
+      .then((res) => res.json())
+      .then((data) => data.data);
+    setDrivers(driverData);
+  };
+
+  useEffect(() => {
+    getExistingDrivers();
+  }, []);
```

Comment:
Is it possible to get the drivers at a higher level, such as when the table renders instead of when the modal  renders? Because I think it sends a request to get all drivers every time you click an assign button? This would involve lifting up the drivers state and passing it down as props so that the request is only made once and the driver dropdowns just use the same data.

Reply:
We just learned about this in Trends today! Raised it to Home because drivers will be used in both Schedule and Table when they're both connected to backend, and Home is their closest ancestor